src/app/datapoints-graph/charts/y-axis.service.ts

```diff
@@ -92,11 +92,20 @@ export class YAxisService {
         axisLabel: {
           fontSize: 10,
           show: !matchingDpRange,
-          formatter: (val) =>
-            new Intl.NumberFormat(this.intlNumberFormatCompliantLocale, {
+          formatter: (val) => {
+            const { min, max } = dp || {};
+            const range =
+              min !== undefined && max !== undefined ? max - min : undefined;
+
+            if (range !== undefined && range < 1) {
```

Comment:
Should we add more smaller ranges? Can we somehow make an algorithm out of this?

`function getDecimalPlaces(num: number): number {
  const numStr = num.toString();
  const decimalIndex = numStr.indexOf('.');
  if (decimalIndex === -1) {
    return 0; // No decimal point found
  }
  return numStr.length - decimalIndex - 1;
}`

Reply:
added